### PR TITLE
Better context information for typed hole errors

### LINKED ISSUE #2165: Show most relevant names in typed hole errors
The context which is shown right now is sometimes helpful, but other times useless, especially when lots of things are in scope. We should compare types to pick the most relevant things to show.

### PR DESCRIPTION
**Pull request by paf31**
Fixes #2165 by only showing "local" things in context, not names of global values/functions. We can add something using the subsumption check later, hopefully.

Also a bit of minor tidying.

### CODE REVIEW

**Comment by paf31** [comment]
Could someone please review this (@kritzcreek maybe, if you have time?), and I'll try to include it in a 0.9.3 release along with the other error message fix.

Thanks.

**Comment by paf31** [comment]
This now includes context info in type wildcard warnings too.

**Comment by garyb** [comment]
Looks good to me, some nice simplifications in `TypeChecker/Types` too now the `ModuleName` lookup can be skipped in a bunch of cases :+1:

File: src/Language/PureScript/Interactive.hs [hunk 1]
```diff
@@ -234,7 +234,7 @@ handleTypeOf val = do
   case e of
     Left errs -> printErrors errs
     Right (_, env') ->
-      case M.lookup (P.ModuleName [P.ProperName "$PSCI"], P.Ident "it") (P.names env') of
+      case M.lookup (P.mkQualified (P.Ident "it") (P.ModuleName [P.ProperName "$PSCI"])) (P.names env') of
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Interactive.hs]: Just something I noticed: Probably a good candidate to extract into a function? Something like `lookupLastResult`?

Comment (paf31) [inline src/Language/PureScript/Interactive.hs]: Yes, good point. I'd like to defer this to another clean-up PR though.

**Pull request merged**